nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -632,22 +627,35 @@ in
       }
     ];
 
+    systemd.tmpfiles.rules = [
+      "d '${cfg.stateDir}' 0750 ${cfg.user} ${cfg.group} - -"
+      "d '${cfg.stateDir}/logs' 0750 ${cfg.user} ${cfg.group} - -"
```

Comment:
You can use `LogsDirectory` from `systemd.exec` instead of this.

Reply:
Unfortunately not:
> The specified directory names must be relative, and may not include "..".

Reply:
Through `LogsDirectory` cannot be specified the full path `/var/spool/nginx`

Reply:
`/var/spool/nginx` directory is not only used to keep logs, but also hashed data and temporary files. On highly loaded systems amount of hashed data may be big. Therefore, placing of them in  `/var/log/nginx` directory is incorrect.

Reply:
Note that `/var/log/nginx/` is the directory used in this [official nginx config example](https://nginx.org/en/docs/example.html). Why do we need to divert from it?

Reply:
I start nginx with the example config:
```
nginx -c /nginx.conf
nginx: [alert] could not open error log file: open() "/nix/store/vz4ryj7my3yzl2xmdvvxax8f97fn8inb-nginx-1.14.0/logs/error.log" failed (30: Read-only file system)
2019/09/04 19:22:25 [emerg] 9461#9461: mkdir() "/nix/store/vz4ryj7my3yzl2xmdvvxax8f97fn8inb-nginx-1.14.0/client_body_temp" failed (30: Read-only file system)
```
Without "-p prefix" switch nginx is trying to write files to a read-only directory /nix/store/4xrqhh1s4xmbnin2d7pkf73hdy1zj9vf-nginx-1.16.1 and fails
```
nginx version: nginx/1.16.1
Usage: nginx [-?hvVtTq] [-s signal] [-c filename] [-p prefix] [-g directives]

Options:
  -?,-h         : this help
  -v            : show version and exit
  -V            : show version and configure options then exit
  -t            : test configuration and exit
  -T            : test configuration, dump it and exit
  -q            : suppress non-error messages during configuration testing
  -s signal     : send signal to a master process: stop, quit, reopen, reload
  -p prefix     : set prefix path (default: /nix/store/4xrqhh1s4xmbnin2d7pkf73hdy1zj9vf-nginx-1.16.1/)
  -c filename   : set configuration file (default: conf/nginx.conf)
  -g directives : set global directives out of configuration file
```

Reply:
With `-p /var/spool/nginx` switch nginx automatically creates subdirectories in the specified directory, or, if it does not have permissions, logs error
`ls -lah /var/spool/nginx` 
```
итого 32K
drwxr-x--- 8 nginx nginx 4,0K авг 13 11:49 .
drwxr-xr-x 6 root  root  4,0K авг 19 20:52 ..
drwx------ 2 nginx nginx 4,0K авг 13 11:49 client_body_temp
drwx------ 4 nginx nginx 4,0K авг 13 21:20 fastcgi_temp
drwxr-x--- 2 nginx nginx 4,0K сен  3 21:34 logs
drwx------ 2 nginx nginx 4,0K авг 13 11:49 proxy_temp
drwx------ 2 nginx nginx 4,0K авг 13 11:49 scgi_temp
drwx------ 2 nginx nginx 4,0K авг 13 11:49 uwsgi_temp
```